merkle.go

```diff
@@ -32,15 +33,56 @@ type Node struct {
 	Right *Node
 }
 
+type HashCountDecorator struct {
```

Comment:
This is only used for testing, right? Could you move it to the tests file?

Reply:
It can be